src/Build/Evaluation/Evaluator.cs

```diff
@@ -636,7 +651,10 @@ private void Evaluate(ILoggingService loggingService, BuildEventContext buildEve
 #endif
                 projectFile = String.IsNullOrEmpty(_projectRootElement.ProjectFileLocation.File) ? "(null)" : _projectRootElement.ProjectFileLocation.File;
 
-                _evaluationLoggingContext = new EvaluationLoggingContext(loggingService, buildEventContext, projectFile);
+                // This will be true when NOT logging property tracking data.
+                if (_evaluationLoggingContext == null)
```

Comment:
Hmm, it's a bit ugly that the logging service and the build event context can now enter either via this method, or via the constructor. Can you move them all to just the constructor and have this method be parameterless?

Reply:
I initially just wanted to touch as few lines of code as possible. I've updated it though so that it all happens in the constructor.